components/tidb_query_expr/src/impl_string.rs

```diff
@@ -679,9 +679,9 @@ pub fn substring_index(
 
 #[rpn_fn]
 #[inline]
-pub fn strcmp(left: BytesRef, right: BytesRef) -> Result<Option<i64>> {
+pub fn strcmp<C: Collator>(left: BytesRef, right: BytesRef) -> Result<Option<i64>> {
     use std::cmp::Ordering::*;
-    Ok(Some(match left.cmp(right) {
+    Ok(Some(match C::sort_compare(left, right).unwrap() {
```

Comment:
Please use `?`. If the error type mismatched, then you need to `.map_err` before the `?`.

Reply:
Maybe we can implement `From` for collection related error?

Reply:
After using `?`, will the error be handled?

Reply:
Yes.

Reply:
It seems that the merge process has not been triggered.